pkg/events/derive/so_export_symbol.go

```diff
@@ -0,0 +1,141 @@
+package derive
+
+import (
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/events/parse"
+	"github.com/aquasecurity/tracee/pkg/utils/sharedobjs"
+	"github.com/aquasecurity/tracee/types/trace"
+	"path"
+	"strings"
+)
+
+func SharedObjectExportWatchedSymbols(soLoader sharedobjs.DynamicSymbolsLoader, watchedSymbols []string, whitelistedLibsPrefixes []string) events.DeriveFunction {
+	gen := initSOExportWatchedSymbolsEventGenerator(soLoader, watchedSymbols, whitelistedLibsPrefixes)
+	return singleEventDeriveFunc(events.SymbolsLoad, gen.GenerateDeriveArgsFunc)
+}
+
+// Most specific paths should be at the top, to prevent bugs with iterations over the list
+var knownLibrariesDirs = []string{
+	"/usr/lib/x86_64-linux-gnu/",
+	"/usr/lib64/",
+	"/usr/lib/",
+	"/lib64/",
+	"/lib/",
+}
+
+// soExportWatchedSymbolsEventGenerator is responsible of generating event if shared object loaded to a process
+// export one or more from given watched sybmols.
+type soExportWatchedSymbolsEventGenerator struct {
+	soLoader            sharedobjs.DynamicSymbolsLoader
+	watchedSymbols      map[string]bool
+	pathPrefixWhitelist []string
+	librariesWhitelist  []string
+}
+
+func initSOExportWatchedSymbolsEventGenerator(
+	soLoader sharedobjs.DynamicSymbolsLoader,
+	watchedSymbols []string,
+	whitelistedLibsPrefixes []string) *soExportWatchedSymbolsEventGenerator {
+	watchedSymbolsMap := make(map[string]bool)
+	for _, sym := range watchedSymbols {
+		watchedSymbolsMap[sym] = true
+	}
+	var libraries, prefixes []string
+	for _, path := range whitelistedLibsPrefixes {
+		if strings.HasPrefix(path, "/") {
+			prefixes = append(prefixes, path)
+		} else {
+			libraries = append(libraries, path)
+		}
+	}
+	return &soExportWatchedSymbolsEventGenerator{
+		soLoader:            soLoader,
+		watchedSymbols:      watchedSymbolsMap,
+		pathPrefixWhitelist: prefixes,
+		librariesWhitelist:  libraries,
+	}
+}
+
+func (soExSymbolGen *soExportWatchedSymbolsEventGenerator) GenerateDeriveArgsFunc(event trace.Event) ([]interface{}, error) {
```

Comment:
This one is not the derive args function, but what creates it.
I think this one should stay the same.

Reply:
If i'm reading it right this function returns arguments, not a function, so it creates/derives the arguments, no?
Also, it can probably be unexported since it's unused outside the package.

Reply:
You are right, I was thinking on previous version of what you see here.
I will fix it.